### PR TITLE
Improve some logging messages for add_kubernetes_metadata processor

### PR DESCRIPTION
**Pull request by flaper87**
- Enhancement

## What does this PR do?

Switch from Debug to Error when unrecoveral events happen and add extra debug messages when indexing and matching pods.

I was trying to debug why my configs wouldn't work and I was forced to enable debug to notice that the `add_kubernetes_metadata` processor was failing because my configs were wrong. 

Sadly, my configs weren't working even after I fixed the above and it was a bit painful to figure out what was going on because there weren't enough debug messages to understand what index keys, matches, and metadata was being processed by filebeat.

## Why is it important?

It makes operations and debugging easier

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by flaper87** [comment]
> @flaper87 Change looks good. Need to add a changelog entry.

Done, thanks for the review :)

File: filebeat/processor/add_kubernetes_metadata/matchers.go [hunk 1]
```diff
@@ -86,7 +86,7 @@ func (f *LogPathMatcher) MetadataIndex(event common.MapStr) string {
 		logp.Debug("kubernetes", "Incoming log.file.path value: %s", source)
 
 		if !strings.Contains(source, f.LogsPath) {
-			logp.Debug("kubernetes", "Error extracting container id - source value does not contain matcher's logs_path '%s'.", f.LogsPath)
+			logp.Err("Error extracting container id - source value does not contain matcher's logs_path '%s'.", f.LogsPath)
```
Comment (kaiyan-sheng) [inline filebeat/processor/add_kubernetes_metadata/matchers.go]: We are in process of refactoring log related like `logp.Err` in https://github.com/elastic/beats/issues/15699. I can create a separate PR for this after this PR gets merged.

Comment (flaper87) [reply filebeat/processor/add_kubernetes_metadata/matchers.go]: thank you! 😊

**Comment by flaper87** [comment]
I can't check the failures in great detail right now. Are these related to the PR or can I send the PR through?

**Comment by blakerouse** [comment]
@flaper87 they seem unrelated

**Comment by exekias** [comment]
backport was missing, I've opened it: https://github.com/elastic/beats/pull/16893

**Pull request merged**